src/style.rs

```diff
@@ -362,34 +477,59 @@ impl Default for Size<Dimension> {
 /// this [introduction to the box model](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Introduction_to_the_CSS_box_model).
 ///
 /// If the behavior does not match the flexbox layout algorithm on the web, please file a bug!
-#[derive(Copy, Clone, PartialEq, Debug)]
+#[derive(Clone, PartialEq, Debug)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde", serde(default))]
 pub struct FlexboxLayout {
     /// What layout strategy should be used?
     pub display: Display,
+
+    // Position properties
     /// What should the `position` value of this struct use as a base offset?
     pub position_type: PositionType,
-    /// Which direction does the main axis flow in?
-    pub flex_direction: FlexDirection,
-    /// Should elements wrap, or stay in a single line?
-    pub flex_wrap: FlexWrap,
-    /// How should items be aligned relative to the cross axis?
-    pub align_items: AlignItems,
-    /// Should this item violate the cross axis alignment specified by its parent's [`AlignItems`]?
-    pub align_self: AlignSelf,
-    /// How should content contained within this item be aligned relative to the cross axis?
-    pub align_content: AlignContent,
-    /// How should items be aligned relative to the main axis?
-    pub justify_content: JustifyContent,
     /// How should the position of this element be tweaked relative to the layout defined?
     pub position: Rect<Dimension>,
+
+    // Size properies
+    /// Sets the initial size of the item
+    // TODO: why does this exist as distinct from flex_basis? How do they interact?
+    pub size: Size<Dimension>,
+    /// Controls the minimum size of the item
+    pub min_size: Size<Dimension>,
+    /// Controls the maximum size of the item
+    pub max_size: Size<Dimension>,
+    /// Sets the preferred aspect ratio for the item
+    ///
+    /// The ratio is calculated as width divided by height.
+    pub aspect_ratio: Option<f32>,
     /// How large should the margin be on each side?
     pub margin: Rect<Dimension>,
     /// How large should the padding be on each side?
     pub padding: Rect<Dimension>,
     /// How large should the border be on each side?
     pub border: Rect<Dimension>,
+
+    // Gap
+    /// How large should the gaps between items in a grid or flex container be?
+    pub gap: Size<Dimension>,
```

Comment:
We should try to get the flexbox gap implemented shortly after this or alongside of it

Reply:
Agreed. Flexbox implementation for `justify_items` and `justify_self` is also missing (see: https://github.com/DioxusLabs/taffy/issues/204#issuecomment-1193202234)